Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,162 @@
+applies for not using MFD in this scenario as it relies on individual function
+devices being physical devices that are DT enumerated.
+
+An example for this kind of requirement is the audio subsystem where a single
+IP handles multiple entities such as HDMI, Soundwire, local devices such as
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The virtual bus is aimed at being a minimal software bus where virtbus
+devices carved out to represent a part of their core device's functionality can
+live on. This bus should be available to be used by any type of device
+(PCI/ACPI/DT). Virtbus devices created by different susbsystems may not share
+much in common with each other. Only the devices created by the same device
+share a common parent and establish a shared communication. The minimalist
+implementation for virtual bus also allows for extensibility by making
+specialized use of the virtbus_device by encapsulating it within other
+structures and adding domain-specific ops.
+
+Virtbus Devices
+~~~~~~~~~~~~~~~
+
+A virbus_device is created and registered to represent a part of its parent
+device's functionality. It is given a match_name that is used for driver
+binding and a release callback that is invoked when the device is unregistered.
+
+.. code-block:: c
+
+	struct virtbus_device {
+		struct device dev;
+		const char *match_name;
+		void (*release)(struct virtbus_device *);
+		u32 id;
+	};
+
+Virtual Bus Drivers
+~~~~~~~~~~~~~~~~~~~
+
+Virtual bus drivers follow the standard driver model convention, where
+discovery/enumeration is handled outside the drivers, and drivers
+provide probe() and remove() methods. They support power management
+and shutdown notifications using the standard conventions.
+
+.. code-block:: c
+
+	struct virtbus_driver {
+		int (*probe)(struct virtbus_device *);
+		int (*remove)(struct virtbus_device *);
+		void (*shutdown)(struct virtbus_device *);
+		int (*suspend)(struct virtbus_device *, pm_message_t);
+		int (*resume)(struct virtbus_device *);
+		struct device_driver driver;
+		const struct virtbus_device_id *id_table;
+	};
+
+Virtual bus drivers register themselves by calling virtbus_register_driver().
+The id_table contains the names of virtbus devices that can bind with the
+driver.
+
+Device Enumeration
+~~~~~~~~~~~~~~~~~~
+
+The virtbus device is enumerated when it is attached to the bus. The device
+is assigned a unique ID automatically that will be appended to its name. If
+two virtbus_devices both named "foo" are registered onto the bus, they will
+have the device names, "foo.x" and "foo.y", where x and y are unique integers.
+
+Driving Binding
+~~~~~~~~~~~~~~~
+
+When a virtual bus device is registered, the driver core iterates through the
```

Comment:
Isnt that how binding happens in driver core? Anything special about virtbus driver to device binding here?

Reply:
Nothing special. I can remove it if its not needed. But it would complete the document